### PR TITLE
Add additional checksum algorithms

### PR DESCRIPTION
**Pull request by mozawa**
Support additional checksum algorithms for object integrity - [HS-54692](https://cloudian.atlassian.net/browse/HS-54692)
Added new additional checksum related headers to handle and updated corresponding internal functions.

[HS-54692]: https://cloudian.atlassian.net/browse/HS-54692?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: boto/s3/checksums.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
+# -*- coding: utf-8 -*-
+
```
Comment (tpodowd) [inline boto/s3/checksums.py]: Is this module your creation or did you find copy some code from elsewhere? If so, you should add the copy right as appropriate.

**Review by tpodowd — COMMENTED** [review]
Question about origin of the checksum code.

File: boto/s3/checksums.py (hunk 1 above)
Comment (mozawa) [reply boto/s3/checksums.py]: Thanks. Added the copy rights to this file https://github.com/cloudian/boto/pull/34/commits/0761a609aae3f40b5c791f7e4c7680f2890b0c21

**Review by tpodowd — APPROVED** [review]
Thanks. Looks good.

**Pull request merged**